modules/oe_migration_views/oe_migration_views.views.inc

```diff
@@ -0,0 +1,468 @@
+<?php
+
+/**
+ * @file
+ * Contains hooks for Views.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\oe_migration_views\Plugin\migrate\id_map\SqlData;
+use Drupal\migrate\Plugin\migrate\destination\EntityContentBase;
+use Drupal\migrate\Plugin\migrate\id_map\Sql;
+use Drupal\migrate\Plugin\Migration;
+
+/**
+ * Implements hook_views_data_alter().
+ */
+function oe_migration_views_views_data_alter(&$data) {
+  $plugin_manager = \Drupal::service('plugin.manager.migration');
+  $migrations = $plugin_manager->createInstances([]);
+
+  /** @var \Drupal\migrate\Plugin\Migration $migration */
+  foreach ($migrations as $migration) {
+    /** @var \Drupal\migrate\Plugin\migrate\id_map\Sql $id_map */
+    $id_map = $migration->getIdMap();
+    if (!is_a($id_map, Sql::class)) {
+      continue;
+    }
+
```

Comment:
This function cause a problem installing the module when others migrate modules are installed (and is more problematic when a lot of migration are defined, you can try with migrate_drupal, for example).

And I think that this is the problem that cause the error installing the site with the docker compose environment. This function increase the size of the $data array, that is saved in cache (and I was investigating but we cannot avoid it) and cause the error in the CacheSet method. I don't have the final solution but I think that this lines can helps:

```
   // Note only for the PR review:  Comment from the README.
    // Views relationships are available to join entity tables if the migrate destination plugin is an `EntityContentBase`.
    if (!is_subclass_of($migration->getDestinationPlugin(), 'Drupal\migrate\Plugin\migrate\destination\Entity')) {
      continue;
    }
```

Basically, as I read in the README this lines remove from the array the content unrelated with EntityContentBase. The problem is that these lines improve the code and reduce the size of $data but it's still failing, we still have tons of lines of $data ready to be saved.

Check the code in ViewsData.php:259, where the view is cached. 

I'm not sure about the requirement of this module and the purpose of expose this huge amount of data into a view, but here is the problem. It's too much and it's a problem saving data in cache due the size of the serialized array.

Reply:
Hi @samuelsolis 
This function is basically exposing the migration tables to views. As there can be a lot of migrations from various modules the size of that array can become quite big.
As for the filtering on migration which use EntityContentBase as destination, it's already [there](https://github.com/openeuropa/oe_migration/blob/D8MT-224/modules/oe_migration_views/oe_migration_views.views.inc#L292).

I made a small fix yesterday in the PR which should reduce the size of the array.
Could you give it a try on your project and see if that makes a difference pls?
If it's not enough I guess I'll have to do something with config to expose only certain migrations to views.

Reply:
Hi again @vever001 

I tried to install the site again (after the new pushed changes) but the module still throwing the same error. Debugging I have notice that the array still have 522 elements and I think that it's difficult to reduce it without some config. I think that we should create a kind of "entities exposed to the migration views" or something like that to filter the number of elements... It's difficult to manage it with tons of data, it's an exponencial growing and difficult to reduce without a "manual" filter.

Reply:
I really don't know the purpose of this module and why we are developing this or if we need it. I asked Ioannis about the requirement of this module (in order to perform a better peer review, and to check if the code is valid for the requirements) but we don't have requirements nor specifications, I don't know why you are the person who has to develop it and no the DMT team (I suppose that we have a reason, for sure, but I don't know it) and really, I don't know nothing. 

I only know that if we enable the module, as you say, with migrate_drupal and something more, the array is huge and the memory limit is not enough, and we can't increase the memory limit (as far as I know). 

Maybe we need to re-define the task before continue.